dbms/src/Databases/DatabaseLazy.cpp

```diff
@@ -361,8 +361,7 @@ StoragePtr DatabaseLazy::loadTable(const Context & context, const String & table
     }
     catch (const Exception & e)
     {
-        throw Exception("Cannot create table from metadata file " + table_metadata_path + ", error: " + e.displayText() +
-            ", stack trace:\n" + e.getStackTrace().toString(),
+        throw Exception("Cannot create table from metadata file " + table_metadata_path + ", error: " + e.displayText() + DB::getCurrentExceptionMessage(true),
```

Comment:
The context of exception is lost.
We must:
- catch by non-const reference and call `e.addMessage` and `throw` (rethrow current exception) to save the stack trace.

Reply:
(this was a small mistake by @nikvas0 that passed review unnoticed)

Reply:
But do we want to preserve the `CANNOT_CREATE_TABLE_FROM_METADATA`?

Perhaps it's better to pass original exception to [this ctor](https://github.com/ClickHouse/ClickHouse/blob/ad2af03f989466a874757583f8bde051ae3238b3/dbms/src/Common/Exception.h#L28), such we have both correct error code and the better stack trace

Reply:
Yes, it can be better.

Reply:
But it's not necessary.